pkg/processor/graph/types.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import "fmt"
+
+type graph struct {
+	vertices map[string]vertex
+}
+
+func newGraph() *graph {
+	return &graph{
+		vertices: make(map[string]vertex),
+	}
+}
+
+func (g *graph) addVertex(name string) {
+	if !g.containsVertex(name) {
+		g.vertices[name] = make(vertex)
+	}
+}
+
+func (g *graph) addEdge(from string, to string) error {
+	f, ok := g.vertices[from]
+	if !ok {
+		return fmt.Errorf("vertex %q not found", from)
+	}
+	_, ok = g.vertices[to]
+	if !ok {
+		return fmt.Errorf("vertex %q not found", to)
+	}
+
+	f.addEdge(to)
+	return nil
+}
+
+func (g *graph) containsVertex(name string) bool {
+	_, ok := g.vertices[name]
+	return ok
+}
+
+type vertex map[string]bool
```

Comment:
If `bool` is only used as a marker (is entry in the map?) then empty struct could be used to save a bit of space and make "used as marker" behaviour explicit: `type vertex map[string]struct{}`